compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -63,6 +92,335 @@ fn check_for_underconstrained_values_within_function(
+                        Value::Intrinsic(intrinsic) => match intrinsic {
+                            Intrinsic::ApplyRangeConstraint | Intrinsic::AssertConstant => {
+                                // Consider these intrinsic arguments constrained
+                                self.clear_constrained(&arguments);
+                            }
+                            Intrinsic::AsWitness | Intrinsic::IsUnconstrained => {
+                                // These intrinsics won't affect the dependency graph
+                            }
+                            Intrinsic::ArrayLen
+                            | Intrinsic::ArrayRefCount
+                            | Intrinsic::ArrayAsStrUnchecked
+                            | Intrinsic::AsField
+                            | Intrinsic::AsSlice
+                            | Intrinsic::BlackBox(..)
+                            | Intrinsic::DerivePedersenGenerators
+                            | Intrinsic::FromField
+                            | Intrinsic::SlicePushBack
+                            | Intrinsic::SlicePushFront
+                            | Intrinsic::SlicePopBack
+                            | Intrinsic::SlicePopFront
+                            | Intrinsic::SliceRefCount
+                            | Intrinsic::SliceInsert
+                            | Intrinsic::SliceRemove
+                            | Intrinsic::StaticAssert
+                            | Intrinsic::StrAsBytes
+                            | Intrinsic::ToBits(..)
+                            | Intrinsic::ToRadix(..)
+                            | Intrinsic::FieldLessThan => {
+                                // Record all the function arguments as parents of the results
+                                self.update_children(&arguments, &results);
+                            }
+                        },
+                        Value::Function(callee) => match all_functions[&callee].runtime() {
+                            RuntimeType::Brillig(_) => {
+                                // Record arguments/results for each Brillig call for the check
+                                trace!("Brillig function {} called at {}", callee, instruction);
+                                self.tainted.insert(
+                                    *instruction,
+                                    BrilligTaintedIds::new(&arguments, &results),
+                                );
+                            }
+                            RuntimeType::Acir(..) => {
+                                // Record all the function arguments as parents of the results
+                                self.update_children(&arguments, &results);
+                            }
+                        },
+                        Value::ForeignFunction(..) => {
+                            debug!("should not be able to reach foreign function from non-Brillig functions, {func_id} in function {}", function.name());
```

Comment:
I'd expect that encountering this would represent a compiler error so it should be a `panic`

Reply:
Ok